CMakeLists.txt

```diff
@@ -4,6 +4,7 @@ project(Cucumber-Cpp)
 
 set(CUKE_USE_STATIC_BOOST ${WIN32} CACHE BOOL "Statically link Boost (except boost::test)")
 set(CUKE_USE_STATIC_GTEST ON CACHE BOOL "Statically link Google Test")
+set(CUKE_ENABLE_SHARED_LIB OFF CACHE BOOL "Generate a shared library")
```

Comment:
Thanks for the comments on this admittedly mangled commit message. 😉 I'll fix and re-push in that branch. However, I just wanted to respond to the `CukeExport.hpp` part -- yes, this was a request from the upstream maintainers. Unfortunately you are correct, this does mean that we'll need to maintain such a file in `build-system`. I've got a branch there with this file (and the necessary GYP changes as well), the branch is named `nre_master_cucumber-cpp-includes`, and I will make a PR for it once we land the static/shared patch in cucumber-cpp's upstream.